lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene60/Lucene60FieldInfosFormat.java

```diff
@@ -311,6 +312,11 @@ private static IndexOptions getIndexOptions(IndexInput input, byte b) throws IOE
     }
   }
 
+  /**
+   * Note: although this format is only used on older versions, we need to keep the write logic
```

Comment:
I hope this assumption is accurate, would appreciate someone double-checking it.

Reply:
In theory it is possible to try to add a new vector field to one segment create with this format, I guess it would break?

Reply:
I don't see how a new vector field would ever be written with this format?  That should use the latest (90) field infos format.

Rather, I think the case that requires writing is when user opens old index/segment, that had used this `FieldInfosFormat`, and then user updates some doc values in the docs in this segment, and then `IndexWriter` needs to write that new doc values field, along with a new `.fnm` (field infos) file.

Reply:
Thanks @mikemccand for the clarification.

I was thinking what would happen if a user opens an old index/segment and adds a new document that contains a vector field.  You are suggesting that in that case a totally new segment would be created. This method is only called in the case you mention above.

Reply:
YW @iverase.

Right, that case should write a whole new segment, using the latest Codec, holding the just indexed document with vector field.